scanner.go

```diff
@@ -196,6 +196,9 @@ func NewScannerFromZipReader(zipReader *zip.Reader, options *ReadShapefileOption
 	var shxFiles []*zip.File
 	var shpFiles []*zip.File
 	for _, zipFile := range zipReader.File {
+		if strings.HasPrefix(zipFile.Name, "__MACOSX") {
```

Comment:
This test is incorrect:

* It does not check for path components, e.g. the directory `__MACOSXABC` also matches, when it shouldn't.
* This does not handle the case of `__MACOSX` directories in subdirectories.

Use `path.Split` instead and check all path components.